test/e2e/operator_test.go

```diff
@@ -1076,36 +1076,47 @@ func TestContainerLogging(t *testing.T) {
 }
 
 func TestIngressControllerCustomEndpoints(t *testing.T) {
-	// TODO: Change from spec to status after https://bugzilla.redhat.com/show_bug.cgi?id=1850681
-	//       is fixed.
-	platform := infraConfig.Spec.PlatformSpec
+	platform := infraConfig.Status.PlatformStatus
+	if platform == nil {
+		t.Fatalf("platform status is missing for infrastructure %s", infraConfig.Name)
+	}
 	switch platform.Type {
 	case configv1.AWSPlatformType:
-		if len(platform.AWS.ServiceEndpoints) != 0 {
+		switch {
+		case platform.AWS == nil:
+			t.Fatalf("aws platform status is missing for infrastructure %s", infraConfig.Name)
+		case len(platform.AWS.ServiceEndpoints) != 0:
 			t.Fatalf("custom endpoints detected for infrastructure %s", infraConfig.Name)
+		case len(platform.AWS.Region) == 0:
+			t.Fatalf("region is missing from aws platform status for infrastructure %s", infraConfig.Name)
 		}
```

Comment:
Would it make sense to add another case to skip if the region is AWS China? Alternatively, we could check that the region is one of the US regions and skip otherwise.